### DISCUSSION TITLE
I have too many conditions to run a function. is there any way I can simplify it with ts-pattern?

### DISCUSSION THREAD

**Discussion by raviteja83** [gvergnaud/ts-pattern#226] [Q&A] [answered]
Is there a way in ts-pattern where I can eliminate a propery/value in each step and if I reach the final step, run whatever fn I want to.

Here is an example:
`
  if(a && b && c && d && e) {
    fn();
  }
`
What would be the recommended way to simplify such conditions?
I can put in array and do `.every`. any alternatives would be much appreciated. 

Thanks for the great work on the library.

**Comment by gvergnaud** [comment, marked as answer]
You can pass several patterns to `.with`, or use `P.intersection(...patterns)`

https://github.com/gvergnaud/ts-pattern?tab=readme-ov-file#matching-several-patterns

https://github.com/gvergnaud/ts-pattern?tab=readme-ov-file#pintersection-patterns
